src/main/java/org/apache/ibatis/migration/commands/NewCommand.java

```diff
@@ -35,35 +44,87 @@ public void execute(String... params) {
-    } else {
+
+    Hook hook = createNewMigrationHook();
+
+    Properties variables = getVariables();
+    variables.setProperty("description", params[0]);
+
+
+    Map<String, Object> hookBindings = createBinding(params);
+    {
+
+      Reader templateReader = getTemplateReader();
+      Change change = (Change) hookBindings.get("change");
+      hook.before(hookBindings);
+      File changeFile = new File(change.getFilename());
       try {
-        String customConfiguredTemplate = getPropertyOption(CUSTOM_NEW_COMMAND_TEMPLATE_PROPERTY);
-        if (customConfiguredTemplate != null) {
-          copyExternalResourceTo(migrationsHome() + "/" + customConfiguredTemplate,
-              Util.file(paths.getScriptPath(), filename), variables);
-        } else {
-          copyDefaultTemplate(variables, filename);
-        }
-      } catch (FileNotFoundException e) {
-        printStream
-            .append("Your migrations configuration did not find your custom template.  Using the default template.");
-        copyDefaultTemplate(variables, filename);
+        copyTemplate(templateReader, changeFile, variables);
+      } catch (IOException e) {
+        throw new MigrationException("Unable to create template file " + changeFile.getAbsolutePath());
       }
+
+      hook.after(hookBindings);
     }
     printStream.println("Done!");
     printStream.println();
+
+  }
+
+  private Reader getTemplateReader() {
+    String def = "org/apache/ibatis/migration/template_migration.sql";
+    Reader templateReader = null;
+    try {
+      templateReader = Resources.getResourceAsReader(def);
+    } catch (IOException e) {
+      throw new MigrationException(String.format("Default template %s can't be found?! ", def), e);
+    }
+    try {
+      String template = getTemplateFile();
+      if (template != null)
+        templateReader = new FileReader(template);
+
+    } catch (FileNotFoundException e) {
+      String msg = String.format(
+          "Your migrations configuration did not find your custom template: %s.  Using the default template.",
+          e.getMessage());
+      printStream.append(msg);
+    }
+    return templateReader;
+  }
+
+  private String getTemplateFile() throws FileNotFoundException {
+    String template = null;
+    if (options.getTemplate() != null) {
+      template = options.getTemplate();
+    } else {
+      String customConfiguredTemplate = getPropertyOption(CUSTOM_NEW_COMMAND_TEMPLATE_PROPERTY);
+      if (customConfiguredTemplate != null) {
+        template = migrationsHome() + "/" + customConfiguredTemplate;
+      }
+    }
+    return template;
   }
 
-  private void copyDefaultTemplate(Properties variables, String filename) {
-    copyResourceTo("org/apache/ibatis/migration/template_migration.sql", Util.file(paths.getScriptPath(), filename),
-        variables);
+  private Map<String, Object> createBinding(String[] params) {
```

Comment:
functionalizes the creation of all binding properties for clarity sake. By functionalizing it the code essentially self documents.